### ISSUE TITLE
How could I check If averaging gradients is running correctly?

### ISSUE THREAD

**Issue by windson87** [horovod/horovod#977]
**Environment:**
1. Framework: Pytorch
2. Framework version: 1.0.0
3. Horovod version:0.16.1
4. MPI version:(IntelMPI)
5. CUDA version:10
7. Python version:3.6.5

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
   Yes
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.md)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.md)?

**Your question:**
Hi, am using horovod to train my deep learning model (a restnet-unet). I have a huuuge dataset (3-4 million images). so instead of using pytorch datasampler, I simply partitioned the data and pre-package for each rank, so that each rank will access ONLY their own package for training (Had issue when multiple ranks trying to iteratively read images from one position, I have around 100 nodes to run in parallel). So each rank will run their own miniset (of course randomness exists in the batch of each rank).

My question is: How can I check if horovod is averaging the gradients correctly? 
The training loss and validation loss look fine, but since am not using pytorch datasampler and dataloader, will it affect the communication between the ranks?

Thanks in advance!!
